client/state/comments/test/actions.js

```diff
@@ -0,0 +1,409 @@
+		} );
+
+		it( 'should not change comments tree when there are no changes available', function( done ) {
+			this.timeout( 5000 );
+
+			const dispatchSpy = sinon.spy();
+
+			const getStateStub = sinon.stub().returns( {
+				comments: {
+					earliestCommentDate: Immutable.fromJS( { [ createCommentTargetId( FEW_COMMENTS_POST.siteId, FEW_COMMENTS_POST.postId ) ]: earliestDate } ),
+					items: Immutable.Map().set( createCommentTargetId( FEW_COMMENTS_POST.siteId, FEW_COMMENTS_POST.postId ), commentsTree )
+				}
+			} );
+
+			pollComments( FEW_COMMENTS_POST.siteId, FEW_COMMENTS_POST.postId )( dispatchSpy, getStateStub ).then( () => {
+				expect( dispatchSpy ).to.not.have.been.called;
+				done();
+			} ).catch( done );
+		} );
+
+		it( 'should fetch new comment if it available', function( done ) {
+			this.timeout( 5000 );
+			const dispatchSpy = sinon.spy();
+
+			const getStateStub = sinon.stub().returns( {
+				comments: {
+					earliestCommentDate: Immutable.fromJS( { [ createCommentTargetId( FEW_COMMENTS_POST.siteId, FEW_COMMENTS_POST.postId ) ]: earliestDate } ),
+					items: Immutable.Map().set( createCommentTargetId( FEW_COMMENTS_POST.siteId, FEW_COMMENTS_POST.postId ), commentsTree.delete( latestDateCommentID ) )
+				}
+			} );
+
+			pollComments( FEW_COMMENTS_POST.siteId, FEW_COMMENTS_POST.postId )( dispatchSpy, getStateStub ).then( () => {
+				expect( dispatchSpy ).to.have.been.called;
+
+				const action = dispatchSpy.args[0][0];
+
+				expect( action.type ).to.equal( COMMENTS_RECEIVE );
+				expect( action.comments.length ).to.equal( 1 );
+				expect( action.comments[0].ID ).to.equal( latestDateCommentID );
+				expect( action.siteId ).to.equal( FEW_COMMENTS_POST.siteId );
+				expect( action.postId ).to.equal( FEW_COMMENTS_POST.postId );
+
+				done();
+			} ).catch( done );
+		} );
+
+		it( 'should remove comment if it no longer on the server', function( done ) {
+			this.timeout( 5000 );
+
+			const dispatchSpy = sinon.spy();
+			const removeCommentId = 123;
+
+			let treeWithExtraComment = commentsTree.set( removeCommentId, Immutable.fromJS( { children: [], data: { date: new Date() } } ) );
+
+			const getStateStub = sinon.stub().returns( {
+				comments: {
+					earliestCommentDate: Immutable.fromJS( { [ createCommentTargetId( FEW_COMMENTS_POST.siteId, FEW_COMMENTS_POST.postId ) ]: earliestDate } ),
+					items: Immutable.Map().set( createCommentTargetId( FEW_COMMENTS_POST.siteId, FEW_COMMENTS_POST.postId ), treeWithExtraComment )
+				}
+			} );
+
+			pollComments( FEW_COMMENTS_POST.siteId, FEW_COMMENTS_POST.postId )( dispatchSpy, getStateStub ).then( () => {
+				expect( dispatchSpy ).to.have.been.called;
+
+				const action = dispatchSpy.args[0][0];
+
+				expect( action.type ).to.equal( COMMENTS_REMOVE_COMMENT );
+				expect( action.commentId ).to.equal( removeCommentId );
+				expect( action.siteId ).to.equal( FEW_COMMENTS_POST.siteId );
+				expect( action.postId ).to.equal( FEW_COMMENTS_POST.postId );
+
+				done();
```

Comment:
I've [refactored](https://github.com/Automattic/wp-calypso/pull/3143/files#diff-7683b0e3a8968ec7d9b075cd827dc5daR181) the async tests to return promises.